engine/apps/api/permissions.py

```diff
@@ -310,18 +350,23 @@ def has_object_permission(self, request: AuthenticatedRequest, view: ViewSetOrAP
 
 
 ALL_PERMISSION_NAMES = [perm for perm in dir(RBACPermission.Permissions) if not perm.startswith("_")]
-ALL_PERMISSION_CLASSES = [
+ALL_PERMISSION_CLASSES: LegacyAccessControlCompatiblePermissions = [
     getattr(RBACPermission.Permissions, permission_name) for permission_name in ALL_PERMISSION_NAMES
 ]
-ALL_PERMISSION_CHOICES = [
+ALL_PERMISSION_CHOICES: typing.List[typing.Tuple[str, str]] = [
     (permission_class.value, permission_name)
     for permission_class, permission_name in zip(ALL_PERMISSION_CLASSES, ALL_PERMISSION_NAMES)
 ]
 
 
-def get_permission_from_permission_string(perm: str) -> typing.Optional[LegacyAccessControlCompatiblePermission]:
+def get_permission_from_permission_string(
+    organization: "Organization", perm: str
+) -> typing.Optional[LegacyAccessControlCompatiblePermission]:
     for permission_class in ALL_PERMISSION_CLASSES:
-        if permission_class.value == perm:
+        permission_class_value = permission_class.value
+        irm_permission_value = convert_oncall_permission_to_irm(permission_class)
+
+        if permission_class_value == perm or organization.is_grafana_irm_enabled and irm_permission_value == perm:
```

Comment:
Precedence works ok here, but maybe add parenthesis grouping the and conditions to make it easier to read?

Reply:
decided to get rid of this method in favour of [ALL_PERMISSION_NAME_TO_CLASS_MAP](https://github.com/grafana/oncall/pull/5149/files#diff-83058af8678d629be3d878a5947537c15c01abf6ef2eb7ba31d08b8febda0853R362-R367)